### PR TITLE
Category/Tag UI bugs

### PR DESCRIPTION
**Pull request by andrewhuang**
**Changes:**
- Moved component declarations above respective controllers (easier to make changes to dependency injection headers)
- Removed "Weekly" from non-cumulative category/tag lists
- Don't update hours when editing categories in non-cumulative category list (since it by default returns cumulative hours)
- misc. changes and nits

### CODE REVIEW

File: cal/cal/static/js/app.js [hunk 1]
```diff
@@ -661,7 +656,6 @@ analyticsApp.controller('CalendarCtrl', function CalendarCtrl($scope, $http, $q,
         right: 'agendaDay,agendaWeek,month today prev,next'
       },
       firstDay: 1,
-      eventClick: $scope.alertOnEventClick,
```
Comment (jamesuejio) [inline cal/cal/static/js/app.js:664]: Why did you delete this?

File: cal/cal/static/js/calendaranalytics.js [hunk 2]
```diff
@@ -26,7 +26,6 @@ $(document).ready(function() {
       data: {
       },
       success: function(data) {
-        tags = [];
```
Comment (jamesuejio) [inline cal/cal/static/js/calendaranalytics.js:29]: I assume deleted because unused?

File: cal/cal/static/js/app.js [hunk 3]
```diff
@@ -326,11 +337,7 @@ function CategoryListCtrl($scope, $http, CalendarFilterService, CategoryService)
 
   this.hideZeroHoursFilter = function(value, index, array) {
     /* jshint unused:vars */
-    if (this.hideZeroHours && value.hours === 0) {
-      return false;
-    } else {
-      return true;
-    }
+    return !this.hideZeroHours || value.hours !== 0
```
Comment (jamesuejio) [inline cal/cal/static/js/app.js]: I think there was a reason why we keep it as an expanded if else. The one liner is a bit hard to understand. In English, not hours or hours not equal 0 is pretty confusing...

File: cal/cal/static/js/app.js [hunk 4]
```diff
@@ -352,7 +359,6 @@ function CategoryListCtrl($scope, $http, CalendarFilterService, CategoryService)
     CategoryService.editCategory(categoryId, category.newLabel)
       .then(function(returnedCategory) {
         category.label = returnedCategory.label;
-        category.hours = returnedCategory.hours;
```
Comment (jamesuejio) [inline cal/cal/static/js/app.js:355]: Does this work on your computer? When I click edit then click save, it still gives me cumulative. Also when I add keywords it gives me cumulative. Try all edge cases and let me know if they all work for you.

**Review by jamesuejio — CHANGES_REQUESTED** [review]
Some questions

File: cal/cal/static/js/app.js (hunk 3 above)
Comment (cindytung) [reply cal/cal/static/js/app.js]: alternatively just do return !(this.hideZeroHours && values.hours === 0)? Even tho it's unsimplified, it's still a one-liner and easy enough to understand

File: cal/cal/static/js/app.js (hunk 1 above)
Comment (andrewhuang) [reply cal/cal/static/js/app.js:664]: The only place `alertOnEventClick` is in there. Once I remove it, I don't see any other mentions.
```
[andrew@Tiffanys-Air-2 calendar-analytics]$ git grep "alertOnEventClick"
[andrew@Tiffanys-Air-2 calendar-analytics]$ 
```

File: cal/cal/static/js/app.js (hunk 4 above)
Comment (andrewhuang) [reply cal/cal/static/js/app.js:355]: This is only for categories, not tags. The tag work will be in another PR.

File: cal/cal/static/js/calendaranalytics.js (hunk 2 above)
Comment (andrewhuang) [reply cal/cal/static/js/calendaranalytics.js:29]: Yes.

**Comment by andrewhuang** [comment]
I've made the changes Cindy recommended to the `hideZeroHoursFilter`s. Let me know if there are other things I need to address.

Also, what’s the convention between using `_this` vs. binding in this codebase? I see some places where both are used.

File: cal/cal/static/js/app.js (hunk 1 above)
Comment (jamesuejio) [reply cal/cal/static/js/app.js:664]: Ok cool

**Review by jamesuejio — APPROVED** [review]
